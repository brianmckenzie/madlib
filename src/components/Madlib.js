import React, { Component } from 'react';

class Madlib extends Component {
    render() {
      return (
        <div className="Madlib-text">
          <p>
            It was a {this.props.word1} and {this.props.word2} night. Suddenly, a {this.props.word3} rang out! <br/><br/>
            A {this.props.word4} slammed. {this.props.word5} screamed.<br/><br/> 
            Suddenly, {this.props.word6} appeared on the horizon! 
          </p>
        </div>
      );
    }
  }
  
  export default Madlib;
  