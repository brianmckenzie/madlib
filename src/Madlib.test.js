import React from 'react';
import ReactDOM from 'react-dom';
import Madlib from './components/Madlib';
import renderer from 'react-test-renderer';

const defaultWords = {
    word1: 'dark',
    word2: 'stormy',
    word3: 'shot',
    word4: 'door',
    word5: 'The maid',
    word6: 'a pirate ship'
};

const changedWords = {
    word1: 'hot',
    word2: 'foul-smelling',
    word3: 'Kanye West Song',
    word4: 'tanning lotion bottle lid',
    word5: 'Stormy Daniels',
    word6: 'an orange walrus'
};

test('Default words return expected text', () => {
    // Render the Madlib with default props and make sure it matches snapshot
    const component = renderer.create(
        <Madlib word1={defaultWords.word1} 
        word2={defaultWords.word2} 
        word3={defaultWords.word3} 
        word4={defaultWords.word4} 
        word5={defaultWords.word5} 
        word6={defaultWords.word6}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // Change one of the props and make sure it no longer matches the snapshot
    tree.props.word1 = "sunny";
    expect(tree).toThrowErrorMatchingSnapshot();
});

test('Changed words return expected text', () => {
    // Render the Madlib with words changed and make sure it matches snapshot
    const component = renderer.create(
        <Madlib word1={changedWords.word1} 
        word2={changedWords.word2} 
        word3={changedWords.word3} 
        word4={changedWords.word4} 
        word5={changedWords.word5} 
        word6={changedWords.word6}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // Change one of the props and make sure it no longer matches the snapshot
    tree.props.word1 = "cold";
    expect(tree).toThrowErrorMatchingSnapshot();
});