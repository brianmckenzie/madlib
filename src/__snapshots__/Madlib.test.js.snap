// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changed words return expected text 1`] = `
<div
  className="Madlib-text"
>
  <p>
    It was a 
    hot
     and 
    foul-smelling
     night. Suddenly, a 
    Kanye West Song
     rang out! 
    <br />
    <br />
    A 
    tanning lotion bottle lid
     slammed. 
    Stormy Daniels
     screamed.
    <br />
    <br />
    Suddenly, 
    an orange walrus
     appeared on the horizon!
  </p>
</div>
`;

exports[`Changed words return expected text 2`] = `"received is not a function"`;

exports[`Default words return expected text 1`] = `
<div
  className="Madlib-text"
>
  <p>
    It was a 
    dark
     and 
    stormy
     night. Suddenly, a 
    shot
     rang out! 
    <br />
    <br />
    A 
    door
     slammed. 
    The maid
     screamed.
    <br />
    <br />
    Suddenly, 
    a pirate ship
     appeared on the horizon!
  </p>
</div>
`;

exports[`Default words return expected text 2`] = `"received is not a function"`;
